src/js/wrapper.ts

```diff
@@ -395,10 +397,25 @@ export const NATIVE: SentryNativeWrapper = {
       throw this._NativeClientError;
     }
 
-    // we stringify the extra as native only takes in strings.
-    const stringifiedExtra = typeof extra === 'string' ? extra : JSON.stringify(extra);
+    if (typeof extra === 'string') {
+      return RNSentry.setExtra(key, extra);
+    }
+    if (typeof extra === 'undefined') {
+      return RNSentry.setExtra(key, 'undefined');
+    }
+
+    let stringifiedExtra: string | undefined;
+    try {
+      const normalizedExtra = normalize(extra);
+      stringifiedExtra = JSON.stringify(normalizedExtra);
+    } catch (e) {
+      logger.error('Extra not passed to native SDK, because it contains non-stringifiable values', e);
```

Comment:
Q: Should we also `return` here or let it fall-through and set `**non-stringifiable**` ?

Reply:
I let it fall thru to `**non-stringifiable**` so the extra doesn't silently disappear from the native events.